backend/core/src/ami-charts/dto/ami-chart.dto.ts

```diff
@@ -1,51 +1,55 @@
 import { Expose, Type } from "class-transformer"
-import { IsDate, IsDefined, IsOptional, IsUUID, ValidateNested } from "class-validator"
-import { OmitType } from "@nestjs/swagger"
+import { IsDate, IsDefined, IsOptional, IsString, IsUUID, ValidateNested } from "class-validator"
 import { AmiChart } from "../entities/ami-chart.entity"
 import { AmiChartItem } from "../entities/ami-chart-item.entity"
 import { ValidationsGroupsEnum } from "../../shared/types/validations-groups-enum"
 import { JurisdictionDto } from "../../jurisdictions/dto/jurisdiction.dto"
 import { IdDto } from "../../shared/dto/id.dto"
+import { HasKeys } from "../../shared/types/has-keys"
+import { AbstractEntity } from "../../shared/entities/abstract.entity"
+
+export class AmiChartDto implements HasKeys<AmiChart> {
```

Comment:
@pbn4 , what's the benefit of using this HasKeys type over `@nestjs/swagger`'s `PickType`? In this case, using PickType, would look like:

```
export class AmiChartDto extends PickType(AmiChart, [
  "id",
  "createdAt",
  "updatedAt",
  "items",
  "name",
] as const) {
  @Expose()
  @Type(() => IdDto)
  jurisdiction: IdDto
}
```

This generates the same interface and still allows us to easily see what's actually being used.

Reply:
@seanmalbert I removed all serialization / validation related properties from AmiChart entity definition and PickType is inheritance based:

```
import { Type } from '@nestjs/common';
import { MappedType } from './mapped-type.interface';
import {
  inheritPropertyInitializers,
  inheritTransformationMetadata,
  inheritValidationMetadata,
} from './type-helpers.utils';

export function PickType<T, K extends keyof T>(
  classRef: Type<T>,
  keys: readonly K[],
): MappedType<Pick<T, typeof keys[number]>> {
  const isInheritedPredicate = (propertyKey: string) =>
    keys.includes(propertyKey as K);

  abstract class PickClassType {
    constructor() {
      inheritPropertyInitializers(this, classRef, isInheritedPredicate);
    }
  }
  inheritValidationMetadata(classRef, PickClassType, isInheritedPredicate);
  inheritTransformationMetadata(classRef, PickClassType, isInheritedPredicate);

  return PickClassType as MappedType<Pick<T, typeof keys[number]>>;
}
```

so defining it like you showed wouldn't work for e.g. input because validation decorators are missing in entity definition. `HasKeys` uses `implements interface` syntax so actually forces you to define AmiChart properties and you then must provide serialization / validation decorators too. It just moves serialization / validation to where it is actually used - in DTOs, and does not clutter entity definition which unnecessary decorators. It's just a way of strongly typing DTO - making sure nothing is missed so it's basically like OmitType but more explicit (no inheritance). What do you think?